Source/Diagnostics/ComputeDiagFunctors/RhoFunctor.cpp

```diff
@@ -0,0 +1,60 @@
+#include "WarpX.H"
+#include "RhoFunctor.H"
+#include "Utils/CoarsenIO.H"
+
+RhoFunctor::RhoFunctor ( const amrex::MultiFab* const mf_src, const int lev,
+                         const amrex::IntVect crse_ratio, bool convertRZmodes2cartesian,
+                         const int ncomp )
+    : ComputeDiagFunctor(ncomp, crse_ratio), m_lev(lev),
+      m_convertRZmodes2cartesian(convertRZmodes2cartesian)
+{
+    // mf_src should not be used, let's make sure it is null
+    AMREX_ALWAYS_ASSERT(mf_src == nullptr);
+    // Write only in one output component
+    AMREX_ALWAYS_ASSERT(ncomp == 1);
```

Comment:
This assert should not be done since it is valid to request writing out all modes. See my suggested changes in FullDiagnostics.cpp.